clarity_ext/repository/step_repository.py

```diff
@@ -169,3 +173,168 @@ def get_process_type(self):
     def get_process(self):
         """Returns the currently running process (step)"""
         return self.session.current_step
+
+
+class WrappablePairs(object):
+    """
+    Contains all artifact pairs within a step, containing genologics Artifact instances
+    These are in turn ready to be wrapped into clarity-ext domain instances
+    Usage:
+    wrappable_pairs = WrappablePairs(session, input_output_maps, mode=<stateful | stateless>)
+    wrappable_pairs.validate()  # compares xml for stateful and stateless
+    for input, output, output_generation_type in wrappable_pairs:
+        input  # genologics Artifact, either stateful or stateless
+        output  # genologics Artifact
+
+    The validation compares the stateless and statefull xml representation,
+    and discards any discrepancies in the qc-flag
+    """
+
+    MODE_STATELESS = "stateless"
+    MODE_STATEFUL = "stateful"
+
+    def __init__(self, session, input_output_maps, state_mode):
+        # Note, the force flag should be set to False as soon as the test period is over!
+        # Develop-1177
+        self.force = True
+        self.session = session
+        self.input_output_maps = input_output_maps
+        self.state_mode = state_mode
+        self.pairs = list()
+        stateful_artifacts = self._fetch_stateful()
+        stateless_artifacts = self._fetch_stateless()
+
+        self.pairs = self._assemble_pairs(stateful_artifacts, stateless_artifacts)
+
+    def validate(self):
+        errors = list()
+        for pair in self.pairs:
+            errors.extend(pair.validate(self.session.api))
+        return errors
+
+    def __iter__(self):
+        self.counter = 0
+        return self
+
+    def next(self):
```

Comment:
I forget how this is with iteration classes. Are you supposed to implement `next`?

Reply:
It seems on this answer on SO that it's not required: https://stackoverflow.com/a/24377/282024

I might be missing something though :)

Reply:
I believe next() is needed in python 2, but not in python 3